Makefile

```diff
@@ -0,0 +1,106 @@
+all: release
+
+# count says how many times to run the tests.
+count = 1
+# pkgs changes which packages the makefile calls operate on. run changes which
+# tests are run during testing.
+pkgs = ./api ./build ./database ./user
+
+# integration-pkgs defines the packages which contain integration tests
+integration-pkgs = ./test
+
+# fmt calls go fmt on all packages.
+fmt:
+	gofmt -s -l -w $(pkgs)
+
+# vet calls go vet on all packages.
+# We don't check composite literals because we need to use unkeyed fields for
+# MongoDB's BSONs and that sets vet off.
+# NOTE: go vet requires packages to be built in order to obtain type info.
+vet:
+	go vet -composites=false $(pkgs)
+
+# markdown-spellcheck runs codespell on all markdown files that are not
+# vendored.
+markdown-spellcheck:
+	git ls-files "*.md" :\!:"vendor/**" | xargs codespell --check-filenames
+
+# lint runs golangci-lint (which includes golint, a spellcheck of the codebase,
+# and other linters), the custom analyzers, and also a markdown spellchecker.
+lint: markdown-spellcheck lint-analyze
+	golangci-lint run -c .golangci.yml
+
+# lint-ci runs golint.
+lint-ci:
+# golint is skipped on Windows.
+ifneq ("$(OS)","Windows_NT")
+# Linux
+	go get golang.org/x/lint/golint
+	golint -min_confidence=1.0 -set_exit_status $(pkgs)
+endif
+
+# lint-analyze runs the custom analyzers.
+lint-analyze:
+	analyze -lockcheck -- $(pkgs)
+
+# start-mongo starts a local mongoDB container with no persistence.
+# The first command stops any running testing container but if there is none the
+# error is ignored.
+start-mongo:
+	docker stop skynet-accounts-mongo-test-db 2>/dev/null || true
+	docker run \
+     --rm \
+     --detach \
+     --name skynet-accounts-mongo-test-db \
+     -p 127.0.0.1:37017:27017 \
+     -e MONGO_INITDB_ROOT_USERNAME=admin \
+     -e MONGO_INITDB_ROOT_PASSWORD=ivolocalpass \
+     mongo
+
+stop-mongo:
+	docker stop skynet-accounts-mongo-test-db
+
+# debug builds and installs debug binaries. This will also install the utils.
+debug:
+	go install -tags='debug profile netgo' -ldflags='$(ldflags)' $(pkgs)
+debug-race:
+	GORACE='$(racevars)' go install -race -tags='debug profile netgo' -ldflags='$(ldflags)' $(pkgs)
+
+# dev builds and installs developer binaries. This will also install the utils.
+dev:
+	go install -tags='dev debug profile netgo' -ldflags='$(ldflags)' $(pkgs)
+dev-race:
+	GORACE='$(racevars)' go install -race -tags='dev debug profile netgo' -ldflags='$(ldflags)' $(pkgs)
+
+# release builds and installs release binaries.
+release:
+	go install -tags='netgo' -ldflags='-s -w $(ldflags)' $(release-pkgs)
+release-race:
+	GORACE='$(racevars)' go install -race -tags='netgo' -ldflags='-s -w $(ldflags)' $(release-pkgs)
+release-util:
+	go install -tags='netgo' -ldflags='-s -w $(ldflags)' $(release-pkgs) $(util-pkgs)
+
+test:
+	go test -short -tags='debug testing netgo' -timeout=5s $(pkgs) -run=. -count=$(count)
+test-long: clean fmt vet lint-ci
```

Comment:
I don't think so. The mental division I make is the following:
- `test`: short unit tests that are quickly executable and people should generally run all of them before each push.
- `test-long`: long test which might set up some internal structures and mocks. People should run the ones that are affected by their PR before pushing.
- `test-int`: integration tests which stand up all external services that we rely on, so we can test against them without using any mocks.

We can also merge the last two. I've set them apart for now because it seems useful to me to have the integration tests be separate and outline entire user stories.